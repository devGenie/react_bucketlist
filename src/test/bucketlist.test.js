import React from 'react';
import { mount, shallow } from 'enzyme';
import { stub, spy } from 'sinon';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import 'should';
import 'should-enzyme';
import localStorageMock from '../__mocks__/localStorage';
import bucketlistmock from '../__mocks__/bucketlists-mock';
import BucketList from '../components/bucketlist'
import sinonStubPromise from 'sinon-stub-promise';

sinonStubPromise(sinon)
window.sessionStorage = localStorageMock;

describe("<Bucketlist />",()=>{
	let bucketlistitems;
	let addFunction;
	let editFuncion;
	let deleteFunction;
	let editFunc;
	let fetchMock;
	let bucketlist;
	let editStub;

	beforeEach(()=>{
		//editFunc = stub();
		editStub=stub();
		fetchMock = stub(window,'fetch').returnsPromise().resolves({status:'success'});
		//doneFunction=spy(BucketListItem.prototype,'handleDone');
		//editFuncion=spy(BucketListItem.prototype,'handleEdit');
		addFunction = spy(BucketList.prototype,'handleAddItem');
		deleteFunction = spy(BucketList.prototype,'handleDeleteBucketlist');
		editFuncion = spy(BucketList.prototype,'handleEditBucketlist');
	 	bucketlist = mount(<BucketList formCallback={editStub} data={{id:1,name:'bucketlist',description:'bucketlist'}}
	 						/>).setState({items:[{id:1,name:'Bucket1',
	 								   			complete_status:true,
	 								   			date_completed:'mm/yy'}]}).update();
	})

	afterEach(()=>{
		fetchMock.restore()
		deleteFunction.restore();
		editFuncion.restore();
		addFunction.restore();
	})

	it("renders correctly",()=>{
		expect(toJson(bucketlist)).toMatchSnapshot();
	});

	it('renders items',()=>{
		//console.log(bucketlist.debug())
		expect(bucketlist.find('li').find('.items')).toHaveLength(1)
	});

	it('can be edited',()=>{
		let editButton = bucketlist.find('.editlist');
		expect(editButton).toHaveLength(1);
		editButton.simulate('click');
		expect(editFuncion.called).toBe(true)
	})

	it('can be deleted',()=>{
		let deleteButton = bucketlist.find('.deletelist');
		expect(deleteButton).toHaveLength(1);
		deleteButton.simulate('click');
		expect(deleteFunction.called).toBe(true)
	})

	it('can add an item',()=>{
		let addButton = bucketlist.find('.addItem');
		expect(addButton).toHaveLength(1);
		addButton.simulate('click');
		expect(addFunction.called).toBe(true)
	})
})