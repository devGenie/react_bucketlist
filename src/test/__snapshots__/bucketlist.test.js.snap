// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bucketlist /> renders correctly 1`] = `
<Bucketlist
  data={
    Object {
      "description": "bucketlist",
      "id": 1,
      "name": "bucketlist",
    }
  }
>
  <div
    className="col s12 m6 l3 bucketlist"
  >
    <div
      className="card bk_card z-depth-5"
    >
      <div
        className="card-content"
      >
        <span
          className="card-title"
        >
          <div
            className="listtitle"
          >
            bucketlist
            <div
              className="right"
            >
              <i
                className="material-icons float"
                data-target="item_model"
                onClick={[Function]}
              >
                add
              </i>
              <i
                className="material-icons float"
                onClick={[Function]}
              >
                delete
              </i>
              <i
                className="material-icons float"
                data-target="EditBucketlist"
                onClick={[Function]}
              >
                edit
              </i>
            </div>
          </div>
          <p
            className="about"
          >
            bucketlist
          </p>
        </span>
        <BucketListItems
          bucketlist={1}
          data={
            Array [
              Object {
                "complete_status": true,
                "date_completed": "mm/yy",
                "id": 1,
                "name": "Bucket1",
              },
            ]
          }
          deleteFunc={[Function]}
          editFunc={[Function]}
        >
          <ul
            className="bkitems"
          >
            <BucketListItem
              bucketlist={1}
              data={
                Object {
                  "complete_status": true,
                  "date_completed": "mm/yy",
                  "id": 1,
                  "name": "Bucket1",
                }
              }
              itemIndex={0}
              key="0"
              onDelete={[Function]}
              onEdit={[Function]}
            >
              <li
                className="items"
              >
                <span
                  className="item_name"
                >
                  Bucket1
                </span>
                <div
                  className="right"
                >
                  <SeeMore>
                    <div
                      className="over_flow_button"
                    >
                      <a
                        className="dropdown_toggle"
                      >
                        <i
                          className="material-icons"
                        >
                          more_vert
                        </i>
                      </a>
                      <div
                        className="overflow_content"
                      >
                        <a
                          className="dropdown_icon completeButton"
                          onClick={[Function]}
                        >
                          <i
                            className="material-icons"
                          >
                            check_box
                          </i>
                        </a>
                        <a
                          className="dropdown_icon editButton"
                          data-target="item_edit_model"
                          onClick={[Function]}
                        >
                          <i
                            className="material-icons"
                          >
                            edit
                          </i>
                        </a>
                        <a
                          className="dropdown_icon deleteButton"
                          onClick={[Function]}
                        >
                          <i
                            className="material-icons"
                          >
                            delete
                          </i>
                        </a>
                      </div>
                    </div>
                  </SeeMore>
                </div>
                <span
                  className="complete_status"
                >
                  <div
                    className="success"
                  >
                    <span
                      className="left"
                    >
                      Completed
                    </span>
                    <i
                      className="material-icons right"
                    >
                      check_box
                    </i>
                  </div>
                </span>
              </li>
            </BucketListItem>
          </ul>
        </BucketListItems>
      </div>
      <Paginate
        loader="show"
        next=""
        previous=""
        trigger={[Function]}
      >
        <ul
          className="item-pagination"
        >
          <div
            className="left"
          >
            <img
              className="loader-img show"
              src="../images/loader.gif"
            />
          </div>
          <div
            className="right"
          >
            <li
              className="item-page item-previous left"
              onClick={[Function]}
            >
              <a>
                <i
                  className="material-icons"
                >
                  chevron_left
                </i>
              </a>
            </li>
            <li
              className="item-page item-next right"
              onClick={[Function]}
            >
              <a>
                <i
                  className="material-icons"
                >
                  chevron_right
                </i>
              </a>
            </li>
          </div>
        </ul>
      </Paginate>
    </div>
  </div>
</Bucketlist>
`;
